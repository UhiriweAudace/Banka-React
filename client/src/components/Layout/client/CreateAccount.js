import React, { Component } from 'react'

class CreateAccount extends Component {
  render() {
    return (
    <div class="main create-account">
      <div></div>
      <div>
        <div class="form">
          <h3 class="d-purple">Create New Bank Account</h3>
          <hr />
          <div class="rows">
            <div class="col-1">
              <label for="">National Identity Card Number <span class="is-required">*</span></label><br/>
              <input type="number" name="id" id="id" placeholder="Type your ID Number" /><br/>
              <span id="error-message" class="error">id field is required</span>
            </div>
            <div class="col-1">
              <label for="">Country Residence <span class="is-required">*</span></label><br/>
              <input type="text" name="country-residence" id="country-residence" placeholder="Type your Country Residence" /><br/>
              <span id="error-message" class="error">Lastname field is required</span>
            </div>
          </div>
          <div class="rows">
            <div class="col-1">
              <label for="">Country Of Birth <span class="is-required"></span></label><br/>
              <input type="text" name="countrybirth" id="countrybirth" placeholder="Type your Country of Birth" /><br/>
              <span id="error-message" class="error"></span>
            </div>
            <div class="col-1">
              <label for="">Date of Birth <span class="is-required">*</span></label><br/>
              <input type="date" name="DateOfBirth" id="DateOfBirth" placeholder="Type your Date Of Birth" /><br/>
              <span id="error-message" class="error">DateOfBirth field is required</span>
            </div>
          </div>
          <div class="rows">
            <div class="col-1">
              <label for="">Currency Type <i>(eg: For Rwandan Francs => Rwf)</i> <span class="is-required">*</span></label><br/>
              <input type="text" name="currency" id="currency" placeholder="Type your currency type" /><br/>
              <span id="error-message" class="error">Currency field is required</span>
            </div>
            <div class="col-1">
              <label for="">Phone Number <span class="is-required"></span></label><br/>
              <input type="number" name="phone" id="phone" placeholder="Type your Phone Number" /><br/>
              <span id="error-message" class="error"></span>
            </div>
          </div>
          <div class="rows">
            <div class="col-1">
              <label for="">Address <span class="is-required"></span></label><br/>
              <input type="text" name="address" id="address" placeholder="Type your current address city" /><br/>
              <span id="error-message" class="error-message"></span>
            </div>
            <div class="col-1">
              <label for="">Account Type <span class="is-required">*</span></label><br/>
              <select name="account-type" id="account-type">
                <option value="1" disabled>Select Account type</option>
                <option value="saving">Saving Account</option>
                <option value="current">Current Account</option>
              </select><br/>
              <span id="error-message" class="error-message"></span>
            </div>
          </div>
          <div class="">
            <div class="col-1">
              <button class="btn-1 primary" id="create">Create</button>
            </div>
          </div>
        </div>
      </div>
    </div>
    )
  }
}

export default CreateAccount;
