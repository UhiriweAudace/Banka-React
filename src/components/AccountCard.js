import React from 'react'

const AccountCard = ({account}) => {
  return (
    <div className="t-row" key={account.accountNumber}>
      <label htmlFor="" className={`white ${account.type === 'saving' ? 'lbl-debit' : 'lbl-credit'}`}>{account.type === 'saving' ? 'Saving' :'Current'} Account</label>
      <div className="t-details white">
        <label htmlFor="" className="small"> N<u>o</u> : 00040-06565-10236-{account.accountNumber}</label>
        <label htmlFor="" className="small">currently on this accounts, You have {account.balance} Rwf.</label>
      </div>
      <div className="t-buttons">
        <button className="btn-flat primary white" id="details" data-set={account.accountNumber}>Details</button>
      </div>
    </div>
  );
}

export default AccountCard
