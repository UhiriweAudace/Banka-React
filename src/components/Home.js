import React, { Component } from 'react'; 
import { Link} from 'react-router-dom';
import '../assets/css/style.css';

class Home extends Component {
  render() {
    return (
      <div>
        <div class="main-content">
          <div class="side-1"></div>
          <div class="side-2">
            <div class="center top">
              <h1 class="heading-1 white">Banka</h1>
              <p class="p-1 white">Your Bank Account In Your Hands.</p>
              <p class="p-1 white">Please Visit Our Nearest Branch For Deposit and Withdrawal Operations!</p>
              <Link to='/auth/signin'>
                <button class="btn-1 primary" id="started">
                  Get Started
                </button>
              </Link>
            </div>
          </div>
          <div class="side-3"></div>
        </div>
      </div>
    )
  }
}

export default Home
