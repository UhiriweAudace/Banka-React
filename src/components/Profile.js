import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from "react-router-dom";
import { capitalize } from '../utils/capitilize';
class Profile extends Component {
  componentDidMount() {
    if (this.props.login.isAuthenticated) {
      this.props.history.push('/profile');
    } else {
      window.location.href = '/';
    }
  }
  render() {
    console.log('::::::::::::::::>>>', this.props.login);
    const { login } = this.props;
    return (
      <div>
        <div className='main prof' id='prof'>
          <div></div>
          <div className='profile secondary'>
            <div className='profile-card'>
              <div className='profile-left'>
                <div className='profile-image center'>
                  {/* <!-- <h5 className="heading-5">Your Profile Picture</h5> --> */}
                </div>
                <button className='btn btn-upload'>
                  Upload Your Profile Picture
                </button>
              </div>
              <div className='profile-content'>
                <h3 className='p-bottom'>Personal Details</h3>
                <ul className='no-bullet '>
                  <li className='secondary'>
                    {capitalize(login.user.firstname)} {''}{' '}
                    {capitalize(login.user.lastname)}
                  </li>
                  <li className='secondary'>{login.user.email}</li>
                  <li className='secondary'>(+250) 785166463</li>
                  <li className='secondary'>Kigali, Rwanda</li>
                  <li className='secondary'>#Current Balance 0 Rwf</li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  login: state.login
});
export default connect(mapStateToProps)(Profile);
