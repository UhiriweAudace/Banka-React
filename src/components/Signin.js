import React, { Component } from 'react';
import { connect } from 'react-redux';
import { signin } from '../redux/actions/signin.action';
import '../assets/css/style.css';
import '../assets/css/signup.css';
import { toast } from 'react-toastify';

export class Signin extends Component {
  state = {
    email: '',
    password: ''
  };

  componentDidMount(){
    if (this.props.login.isAuthenticated) {
      this.props.history.push('/profile')
    } 
    this.redirectOnLoginSuccess();
  }
  componentWillReceiveProps(nextProps){
    if (nextProps.login.isAuthenticated) {
      this.props.history.push('/profile')
    }

  }
  onChangeHandler = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };
  onSubmitHandler = e => {
    e.preventDefault();
    const { email, password } = this.state;
    if (email ==='') return toast.warn('Email is required!');
    if (password ==='') return toast.warn('password is required!');
    const data = {
      email,
      password
    };
    this.props.signin(data);
  };

  redirectOnLoginSuccess = () =>{
    const { login } = this.props;
    return login.isAuthenticated ? this.props.history.push('/profile' || '/') : null;
  }
  render() {
    const { login } = this.props;
    if (login.errors !== undefined && login.errors.status === 400) {
      toast.warn(`${login.errors.message}`);
    }
    if (login.errors !== undefined && login.errors.status === 404) {
      toast.warn(`${login.errors.message}`);
    }
    return (
      <main>
        <div className='main-content'>
          <div className='side-1'></div>
          <div className='side-2'>
            <div className='form'>
              <h3 className='d-purple'>Login</h3>
              <hr />
              <div className='rows'>
                <div className='col-1'>
                  <label htmlFor=''>
                    Email <span className='is-required'>*</span>
                  </label>
                  <br />
                  <input
                    type='text'
                    name='email'
                    id='email'
                    placeholder='Type your e-mail'
                    onChange={this.onChangeHandler}
                  />
                  <br />
                  <span id='error-message' className='error'>
                    Email field is required
                  </span>
                </div>
                <div className='col-1'>
                  <label htmlFor=''>
                    Password <span className='is-required'>*</span>
                  </label>
                  <br />
                  <input
                    type='password'
                    name='password'
                    id='Password'
                    placeholder='Type your Password'
                    onChange={this.onChangeHandler}
                  />
                  <br />
                  <span id='error-message' className='error'>
                    Password field is required
                  </span>
                </div>
              </div>
              <div className='rows'>
                <div className='col-1'>
                  <button
                    className='btn-1 primary bottom'
                    id='login'
                    type='submit'
                    onClick={this.onSubmitHandler}
                  >
                    Login
                  </button>{' '}
                  &nbsp;
                  <button className='btn-1 primary' id='reset'>
                    Reset Password
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div className='side-3'></div>
        </div>
      </main>
    );
  }
}
const mapStateToProps = state => ({
  login: state.login
});
export default connect(
  mapStateToProps,
  { signin }
)(Signin);
