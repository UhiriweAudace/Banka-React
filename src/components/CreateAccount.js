import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import PropTypes from 'prop-types';
import { createAccount } from '../redux/actions/createAccount.action';

import '../assets/css/style.css';
import '../assets/css/signup.css';
import '../assets/css/client.css';

class CreateAccount extends Component {
  state = {
    nationalId: '',
    countryResidence: '',
    DateOfBirth: '',
    countryBirth: '',
    currency: '',
    phone: '',
    address: '',
    accountType: 'current'
  };

  onchangeHandler = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  onSubmitHandler = e => {
    e.preventDefault();
    const { accountType, currency, DateOfBirth, nationalId } = this.state;

    if (accountType === '') return toast.warn('Account Type is required!');
    if (currency === '') return toast.warn('Currency is required!');
    if (DateOfBirth === '') return toast.warn('Date Of Birth is required!');
    if (nationalId === '') return toast.warn('National Id is required!');
    const accountData = {
      type: accountType
    };
    this.props.createAccount(accountData);
  };
  componentDidMount() {
    if (!this.props.login.isAuthenticated) {
      window.location.href = '/auth/signin';
    }
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.login.isAuthenticated) {
      this.props.history.push('/accounts/new');
    } else {
      window.location.href = '/auth/signin';
    }

  }

  render() {
    const { account } = this.props;
    if (account !== undefined && account.status === 201) {
      toast.success('   Account created Successfully!');
    }
    console.log('User account type in render::::', this.props);

    return (
      <div className='main create-account' id='create-account'>
        <div></div>
        <div>
          <div className='form'>
            <h3 className='d-purple'>Create New Bank Account</h3>
            <hr />
            <div className='rows'>
              <div className='col-1'>
                <label htmlFor=''>
                  National Identity Card Number{' '}
                  <span className='is-required'>*</span>
                </label>
                <br />
                <input
                  type='number'
                  name='nationalId'
                  id='id'
                  placeholder='Type your ID Number'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'>
                  id field is required
                </span>
              </div>
              <div className='col-1'>
                <label htmlFor=''>
                  Country Residence <span className='is-required'>*</span>
                </label>
                <br />
                <input
                  type='text'
                  name='countryResidence'
                  id='country-residence'
                  placeholder='Type your Country Residence'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'>
                  Lastname field is required
                </span>
              </div>
            </div>
            <div className='rows'>
              <div className='col-1'>
                <label htmlFor=''>
                  Country Of Birth <span className='is-required'></span>
                </label>
                <br />
                <input
                  type='text'
                  name='countryBirth'
                  id='countrybirth'
                  placeholder='Type your Country of Birth'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'></span>
              </div>
              <div className='col-1'>
                <label htmlFor=''>
                  Date of Birth <span className='is-required'>*</span>
                </label>
                <br />
                <input
                  type='date'
                  name='DateOfBirth'
                  id='DateOfBirth'
                  placeholder='Type your Date Of Birth'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'>
                  DateOfBirth field is required
                </span>
              </div>
            </div>
            <div className='rows'>
              <div className='col-1'>
                <label htmlFor=''>
                  Currency Type <i>(eg: For Rwandan Francs => Rwf)</i>{' '}
                  <span className='is-required'>*</span>
                </label>
                <br />
                <input
                  type='text'
                  name='currency'
                  id='currency'
                  placeholder='Type your currency type'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'>
                  Currency field is required
                </span>
              </div>
              <div className='col-1'>
                <label htmlFor=''>
                  Phone Number <span className='is-required'></span>
                </label>
                <br />
                <input
                  type='number'
                  name='phone'
                  id='phone'
                  placeholder='Type your Phone Number'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'></span>
              </div>
            </div>
            <div className='rows'>
              <div className='col-1'>
                <label htmlFor=''>
                  Address <span className='is-required'></span>
                </label>
                <br />
                <input
                  type='text'
                  name='address'
                  id='address'
                  placeholder='Type your current address city'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error-message'></span>
              </div>
              <div className='col-1'>
                <label htmlFor=''>
                  Account Type <span className='is-required'>*</span>
                </label>
                <br />
                <select
                  name='accountType'
                  id='account-type'
                  onChange={this.onchangeHandler}
                >
                  <option value='0' disabled>
                    Select Account type
                  </option>
                  <option value='current'>Current Account</option>
                  <option value='saving'>Saving Account</option>
                </select>
                <br />
                <span id='error-message' className='error-message'></span>
              </div>
            </div>
            <div className=''>
              <div className='col-1'>
                <button
                  className='btn-1 primary'
                  id='create'
                  type='submit'
                  onClick={this.onSubmitHandler}
                >
                  Create
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
CreateAccount.propTypes = {
  login: PropTypes.object.isRequired,
  createAccount: PropTypes.func.isRequired
};
const mapStateToProps = state => ({
  login: state.login,
  account: state.accountData
});

export default connect(
  mapStateToProps,
  { createAccount }
)(CreateAccount);
