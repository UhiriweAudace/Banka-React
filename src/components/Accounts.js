import React, { Component } from 'react';
import { connect } from 'react-redux';
import AccountCard from './AccountCard';
import {toast} from 'react-toastify';
import {getAllAccounts} from '../redux/actions/account.action';

class Accounts extends Component {

  componentDidMount() {
    if (!this.props.login.isAuthenticated) {
      window.location.href = '/auth/signin';
    }
    this.props.getAllAccounts();
  }
  render() {
    const { accounts} = this.props;
    console.log('User account type in render::::', accounts.accounts);
    return (
      <div className="main my-bank-accounts" id="my-bank-accounts">
        <div></div>
        <div className="transaction-list">
          {
            accounts !== undefined && accounts.accounts.data !== undefined ? accounts.accounts.data.map( account => {
              toast.success(`    ${accounts.accounts.message}`)
              return (
                <AccountCard account={account}/>
              );
            }) : (
              <div className="t-row">
                <label htmlFor="" className='white lbl-debit'></label>
                <div className="t-details white">
                  <label htmlFor="" className="small"> </label>
                  <label htmlFor="" className="small">  Whoops! Currently, You don't have any Bank Accounts.</label>
                </div>
                <div className="t-buttons">
                  {/* <button className="btn-flat primary white" id="details" >Details</button> */}
                </div>
              </div>
            )
          }
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  login: state.login,
  accounts: state.account,
})
export default connect(mapStateToProps, { getAllAccounts })(Accounts);
