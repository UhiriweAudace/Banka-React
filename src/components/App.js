import React, { Component } from 'react';
import { BrowserRouter, Route, Switch} from 'react-router-dom';
import Header from './Layout/Header';
import Home from './Home';
import Signin from './Signin';
import Signup from './Signup';


class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div>
          <Header/>
          <Switch>
            <Route exact path='/' component={Home} />
            <Route exact path='/auth/signin' component={Signin} />
            <Route exact path='/auth/signup' component={Signup} />
          </Switch>
        </div>
      </BrowserRouter>
    )
  }
}

export default App
