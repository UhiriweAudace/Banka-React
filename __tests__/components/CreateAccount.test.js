import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CreateAccount } from '../../src/components/CreateAccount';

let wrapper;
describe('Signup Page', () => {
  test('It should render the CreateAccount Page', () => {
    const props = {
      login: {
        isAuthenticated: true
      },
      account: {
        message: 'account created successfully' 
      },
      history: {
        push: jest.fn()
      },
      createAccount: jest.fn()
    };
    wrapper = shallow(<CreateAccount {...props} />);
    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

  test('It should allow users to enter his/her information', () => {
    const id = wrapper.find('#id');
    const countryResidence = wrapper.find('#country-residence');
    const countryBirth = wrapper.find('#countryBirth');
    const DateOfBirth = wrapper.find('#DateOfBirth');
    const currency = wrapper.find('#currency');
    const phone = wrapper.find('#phone');
    const accountType = wrapper.find('#account-type');
    
    id.simulate('change', {
      target: { value: '42352', name: 'nationalId' }
    });
    countryResidence.simulate('change', {
      target: { value: 'Rwanda', name: 'countryResidence' }
    });
    countryBirth.simulate('change', {
      target: { value: 'Rwanda', name: 'countryBirth' }
    });
    DateOfBirth.simulate('change', {
      target: { value: '12-09-1888', name: 'DateOfBirth' }
    });
    currency.simulate('change', {
      target: { value: 'RWf', name: 'currency' }
    });
    phone.simulate('change', {
      target: { value: '6543463', name: 'phone' }
    });
    accountType.simulate('change', {
      target: { value: '00986543463', name: 'accountType' }
    });
    expect(wrapper.state('nationalId')).toEqual('42352');
    expect(wrapper.state('countryResidence')).toEqual('Rwanda');
    expect(wrapper.state('countryBirth')).toEqual('Rwanda');
    expect(wrapper.state('DateOfBirth')).toEqual('12-09-1888');
    expect(wrapper.state('currency')).toEqual('RWf');
    expect(wrapper.state('phone')).toEqual('6543463');
    expect(wrapper.state('accountType')).toEqual('00986543463');
  });
});
describe('submit button test...', () => {
  test('User will be able to click on submit button and send an action', () => {
    let submitButton;
    const preventDefault = jest.fn();
    submitButton = wrapper.find('#create');
    submitButton.simulate('click', { preventDefault });
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(preventDefault).toBeCalled();
  });
});
