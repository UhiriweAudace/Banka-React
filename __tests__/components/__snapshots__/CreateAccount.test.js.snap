// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Page It should render the CreateAccount Page 1`] = `ShallowWrapper {}`;

exports[`submit button test... User will be able to click on submit button and send an action 1`] = `
<div
  className="main create-account"
  id="create-account"
>
  <div />
  <div>
    <div
      className="form"
    >
      <h3
        className="d-purple"
      >
        Create New Bank Account
      </h3>
      <hr />
      <div
        className="rows"
      >
        <div
          className="col-1"
        >
          <label
            htmlFor=""
          >
            National Identity Card Number
             
            <span
              className="is-required"
            >
              *
            </span>
          </label>
          <br />
          <input
            id="id"
            name="nationalId"
            onChange={[Function]}
            placeholder="Type your ID Number"
            type="number"
          />
          <br />
          <span
            className="error"
            id="error-message"
          >
            id field is required
          </span>
        </div>
        <div
          className="col-1"
        >
          <label
            htmlFor=""
          >
            Country Residence 
            <span
              className="is-required"
            >
              *
            </span>
          </label>
          <br />
          <input
            id="country-residence"
            name="countryResidence"
            onChange={[Function]}
            placeholder="Type your Country Residence"
            type="text"
          />
          <br />
          <span
            className="error"
            id="error-message"
          >
            Lastname field is required
          </span>
        </div>
      </div>
      <div
        className="rows"
      >
        <div
          className="col-1"
        >
          <label
            htmlFor=""
          >
            Country Of Birth 
            <span
              className="is-required"
            />
          </label>
          <br />
          <input
            id="countryBirth"
            name="countryBirth"
            onChange={[Function]}
            placeholder="Type your Country of Birth"
            type="text"
          />
          <br />
          <span
            className="error"
            id="error-message"
          />
        </div>
        <div
          className="col-1"
        >
          <label
            htmlFor=""
          >
            Date of Birth 
            <span
              className="is-required"
            >
              *
            </span>
          </label>
          <br />
          <input
            id="DateOfBirth"
            name="DateOfBirth"
            onChange={[Function]}
            placeholder="Type your Date Of Birth"
            type="date"
          />
          <br />
          <span
            className="error"
            id="error-message"
          >
            DateOfBirth field is required
          </span>
        </div>
      </div>
      <div
        className="rows"
      >
        <div
          className="col-1"
        >
          <label
            htmlFor=""
          >
            Currency Type 
            <i>
              (eg: For Rwandan Francs =&gt; Rwf)
            </i>
             
            <span
              className="is-required"
            >
              *
            </span>
          </label>
          <br />
          <input
            id="currency"
            name="currency"
            onChange={[Function]}
            placeholder="Type your currency type"
            type="text"
          />
          <br />
          <span
            className="error"
            id="error-message"
          >
            Currency field is required
          </span>
        </div>
        <div
          className="col-1"
        >
          <label
            htmlFor=""
          >
            Phone Number 
            <span
              className="is-required"
            />
          </label>
          <br />
          <input
            id="phone"
            name="phone"
            onChange={[Function]}
            placeholder="Type your Phone Number"
            type="number"
          />
          <br />
          <span
            className="error"
            id="error-message"
          />
        </div>
      </div>
      <div
        className="rows"
      >
        <div
          className="col-1"
        >
          <label
            htmlFor=""
          >
            Address 
            <span
              className="is-required"
            />
          </label>
          <br />
          <input
            id="address"
            name="address"
            onChange={[Function]}
            placeholder="Type your current address city"
            type="text"
          />
          <br />
          <span
            className="error-message"
            id="error-message"
          />
        </div>
        <div
          className="col-1"
        >
          <label
            htmlFor=""
          >
            Account Type 
            <span
              className="is-required"
            >
              *
            </span>
          </label>
          <br />
          <select
            id="account-type"
            name="accountType"
            onChange={[Function]}
          >
            <option
              disabled={true}
              value="0"
            >
              Select Account type
            </option>
            <option
              value="current"
            >
              Current Account
            </option>
            <option
              value="saving"
            >
              Saving Account
            </option>
          </select>
          <br />
          <span
            className="error-message"
            id="error-message"
          />
        </div>
      </div>
      <div
        className=""
      >
        <div
          className="col-1"
        >
          <button
            className="btn-1 primary"
            id="create"
            onClick={[Function]}
            type="submit"
          >
            Create
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
